plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -0,0 +1,264 @@
+        fmt_str = "Device ID : {0}\n" + \
+                  "Model ID  : {1}\n" + \
+                  "Batt Lvl  : {2}\n" + \
+                  "Sig Str   : {3}"
+        return fmt_str.format(camera['deviceId'],
+                              camera['properties']['modelId'],
+                              camera['batteryLevel'],
+                              camera['signalStrength'])
+
+    def _get_camera_snapshot(self, device_id):
+        if REAL_SETTINGS.getSetting('show_snapshots'):
+            camera = self._get_camera(device_id)
+            snapshot_file = "{0}/resources/media/{1}.jpg".format(ADDON_PATH, camera['deviceId'])
+            snapshot_url = camera['presignedFullFrameSnapshotUrl']
+            #self.log("*** file: {}".format(snapshot_file))
+            #self.log("*** URL : {}".format(snapshot_url))
+            try:
+                self.arlo.DownloadSnapshot(snapshot_url, snapshot_file, 4096)
+            except:
+                self.log("Unable to download snapshot: {0}".format(snapshot_url), xbmc.LOGERROR)
+                snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        else:
+            snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        self.log("  snapshot_file name = {0}".format(snapshot_file), xbmc.LOGDEBUG)
+        return snapshot_file
+
+    def stop_camera(self):
+        pass
+
+    def play_camera(self, device_id):
+        # Send the command to start the stream and return the stream url.
+        camera = self._get_camera(device_id)
+        stream_url = self.arlo.StartStream(self.basestation, camera)
+
+        # Create a playable item with a path to play.
+        play_item = xbmcgui.ListItem(path=stream_url, label=camera['deviceName'])
+        play_item.setProperty("isPlayable", "true")
+        # Pass the item to the Kodi player.
+        xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+
+        wait_secs = 5
+        is_playing = False
+        timed_out = False
+        start_time = time.time()
+        while not (is_playing or timed_out):
+            is_playing = xbmc.Player().isPlaying()
+            if is_playing:
+                self.log("Player has started...", xbmc.LOGDEBUG)
+            else:
+                if time.time() - start_time < wait_secs:
+                    self.log("Waiting for player to start..", xbmc.LOGDEBUG)
+                    time.sleep(1)
+                else:
+                    self.log("Timed out waiting for player to start", xbmc.LOGWARNING)
+                    timed_out = True
+
+        while is_playing:
+            # self.log("Player is playing...", xbmc.LOGDEBUG)
+            time.sleep(1)
+            is_playing = xbmc.Player().isPlaying()
+            if not is_playing:
+                self.log("Player has stopped.", xbmc.LOGDEBUG)
+
+        # Update camera details after viewing (battery strength, signal strength,...)
+        self._update_arlo_cameras_details()
```

Comment:
I don't see a place where the update details are use, as this is the last command executed and it doesn't persist the info.